ballerina-to-openapi/src/main/java/io/ballerina/openapi/service/mapper/ServiceToOpenAPIMapper.java

```diff
@@ -185,6 +188,7 @@ private static void extractServiceNodes(String serviceName, List<String> availab
     public static OASResult generateOAS(OASGenerationMetaInfo oasGenerationMetaInfo) {
         ServiceDeclarationNode serviceDefinition = oasGenerationMetaInfo.getServiceDeclarationNode();
         ModuleMemberVisitor moduleMemberVisitor = extractNodesFromProject(oasGenerationMetaInfo.getProject());
+        extractHateoasLinkMetadata(oasGenerationMetaInfo.getProject());
```

Comment:
Where do we use these extracted HATEOAS details in OAS mapping?

Reply:
Can you refer to [HateoasContextHolder](https://github.com/ballerina-platform/openapi-tools/blob/1461a87e5ca274fd8cc30794499c570a034529b3/ballerina-to-openapi/src/main/java/io/ballerina/openapi/service/mapper/hateoas/HateoasContextHolder.java)